### PR TITLE
Restore compatiblity with junit-sql-storage

### PR DESCRIPTION
**Pull request by timja**
and some minor cleanup at the same time, bunch of unused variables mostly.

Also fixed clicking on builds in the charts, variable wasn't defined properly before

cc @mdealer

noticed in https://github.com/jenkinsci/junit-sql-storage-plugin/pull/430
caused by https://github.com/jenkinsci/junit-plugin/pull/625

### Testing done

manually tested both junit and junit-sql-storage

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/hudson/tasks/junit/History.java [hunk 1]
```diff
@@ -170,30 +163,29 @@ private ObjectNode computeDurationTrendJson(List<HistoryTestResultSummary> histo
...
             domainAxisLabels.add(fdn);
-            ObjectNode durationColor = mapper.createObjectNode();
-            double duration = Math.round(mul * to.getDuration() * roundMul) / roundMul;
+            ObjectNode durationColor = MAPPER.createObjectNode();
+            double duration = Math.round(mul * h.getDuration() * roundMul) / roundMul;
             tmpMax = Math.max((float)duration, tmpMax);
-            lrY[index] = (double)(duration);
+            lrY[index] = duration;
             durationColor.put("value", duration);
-            if (to.isPassed() || (to.getPassCount() > 0 && to.getFailCount() == 0)) {
```
Comment (timja) [inline src/main/java/hudson/tasks/junit/History.java:191]: minor behaviour change here but should be safe enough

File: src/main/java/hudson/tasks/junit/HistoryTestResultSummary.java [hunk 2]
```diff
@@ -11,21 +13,19 @@ public class HistoryTestResultSummary {
     private final int skipCount;
     private final int passCount;
     private final String description;
-    private final hudson.tasks.test.TestResult resultInRun;
 
-    public HistoryTestResultSummary(Run<?, ?> run, hudson.tasks.test.TestResult resultInRun,
+    public HistoryTestResultSummary(Run<?, ?> run,
```
Comment (timja) [inline src/main/java/hudson/tasks/junit/HistoryTestResultSummary.java:17]: this was the critical change, adding `resultInRun` broke the plugin and I can't see a way to retrieve it there. The API could be modified to maybe pass some more info down but it didn't seem necessary the only additional information that is missing now is the id of the test report which could be stored maybe if required

File: src/test/java/io/jenkins/plugins/junit/storage/TestResultStorageJunitTest.java [hunk 3]
```diff
@@ -476,7 +476,7 @@ public List<HistoryTestResultSummary> getHistorySummary(int offset) {
                                     Job<?, ?> theJob = Jenkins.get().getItemByFullName(getJobName(), Job.class);
                                     if (theJob != null) {
                                         Run<?, ?> run = theJob.getBuildByNumber(buildNumber);
-                                        historyTestResultSummaries.add(new HistoryTestResultSummary(run, null, duration, failed, skipped, passed));
```
Comment (timja) [inline src/test/java/io/jenkins/plugins/junit/storage/TestResultStorageJunitTest.java:479]: this shouldn't have had null passed as it made the test run but if any one actually tried to access the history view they got null pointers

File: src/main/java/hudson/tasks/junit/HistoryTestResultSummary.java (hunk 2 above)
Comment (mdealer) [reply src/main/java/hudson/tasks/junit/HistoryTestResultSummary.java:17]: Any reason to not just pass null to it? Without the final URL it's very cumbersome to navigate single test or test subset history.

Comment (timja) [reply src/main/java/hudson/tasks/junit/HistoryTestResultSummary.java:17]: can't remember offhand, would be best to keep the same features though i.e. it shouldn't be harder to navigate the tests when using pluggable storage

Comment (mdealer) [reply src/main/java/hudson/tasks/junit/HistoryTestResultSummary.java:17]: Well, it is rather nearly impossible to navigate with 10K+ tests. This is what a user normally expects when viewing the history at a certain level. After all, the numbers shown are for the test and test set, not the overall build, but the URL is pointing to overall build, so it is a bug depending on the perspective.

Comment (timja) [reply src/main/java/hudson/tasks/junit/HistoryTestResultSummary.java:17]: feel free to send a PR that maintains compatibility

**Pull request merged**